hack/jenkins/test-history/gen_json.py

```diff
@@ -136,17 +144,20 @@ def get_tests_from_junit_path(path):
             continue
         yield test
 
+
 def get_tests_from_build(job, build):
     """Generates all tests for a build."""
     for junit_path in gcs_ls_junit_paths(job, build):
         for test in get_tests_from_junit_path(junit_path):
             yield test
 
-def get_daily_builds(server, prefix):
+
+def get_daily_builds(server, match):
     """Generates all (job, build) pairs for the last day."""
     now = time.time()
+    matcher = re.compile(match)
```

Comment:
put the re.compile in the main function and make get_daily_builds take an arbitrary predicate function?